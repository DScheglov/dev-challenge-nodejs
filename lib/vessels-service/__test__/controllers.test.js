// eslint-disable-next-line global-require
jest.mock('../../models', () => require('./__mock-models__'));
const controllers = require('../controllers');

describe('controllers.getAllVessels', () => {
  it('should call res.json with DB result', async () => {
    const req = {};
    const res = { json: jest.fn() };
    const next = jest.fn();

    await controllers.getAllVessels(req, res, next);

    expect(
      res.json.mock.calls[0][0]
    ).toMatchSnapshot();

    expect(
      next
    ).not.toHaveBeenCalled();
  });
});


describe('controllers.getVesselRequests', () => {
  it('should call res.json with DB result', async () => {
    const req = { parsed: { vessel: 1 } };
    const res = { json: jest.fn() };
    const next = jest.fn();

    await controllers.getVesselRequests(req, res, next);

    expect(
      res.json.mock.calls[0][0]
    ).toMatchSnapshot();

    expect(
      next
    ).not.toHaveBeenCalled();
  });

  it('should call next with 404 error if result is an empty array', async () => {
    const req = { parsed: { vessel: 2 } };
    const res = { json: jest.fn() };
    let err = null;
    const next = error => {
      err = error;
    };

    await controllers.getVesselRequests(req, res, next);

    expect(err.status).toBe(404);
    expect(err.message).toBe('Vessel "2" is not found.');

    expect(
      res.json
    ).not.toHaveBeenCalled();
  });
});
