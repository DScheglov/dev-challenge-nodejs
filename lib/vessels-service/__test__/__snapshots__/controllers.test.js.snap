// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controllers.getAllVessels should call res.json with DB result 1`] = `
Array [
  Object {
    "target": 1,
    "vessel": 1,
  },
  Object {
    "target": 2,
    "vessel": 2,
  },
]
`;

exports[`controllers.getVesselRequests should call res.json with DB result 1`] = `
Array [
  Object {
    "requestDate": 2019-03-22T10:12:12.000Z,
    "result": Array [],
    "target": 2,
    "vessel": 1,
  },
  Object {
    "requestDate": 2019-03-22T10:12:15.000Z,
    "result": Array [],
    "target": 2,
    "vessel": 1,
  },
]
`;
